src/taipy/core/submission/submission.py

```diff
@@ -130,56 +145,46 @@ def __gt__(self, other):
     def __ge__(self, other):
         return self.creation_date.timestamp() >= other.creation_date.timestamp()
 
-    def _update_submission_status(self, _: Job):
-        abandoned = False
-        canceled = False
-        blocked = False
-        pending = False
-        running = False
-        completed = False
-
-        for job in self.jobs:
-            if not job:
-                continue
-            if job.is_failed():
-                self.submission_status = SubmissionStatus.FAILED  # type: ignore
-                return
-            if job.is_canceled():
-                canceled = True
-                continue
-            if job.is_blocked():
-                blocked = True
-                continue
-            if job.is_pending() or job.is_submitted():
-                pending = True
-                continue
-            if job.is_running():
-                running = True
-                continue
-            if job.is_completed() or job.is_skipped():
-                completed = True
-                continue
-            if job.is_abandoned():
-                abandoned = True
-        if canceled:
+    def _update_submission_status(self, job: Job):
```

Comment:
I am concern about the the possibility to call this method for the same job for 2 (or more) status changes in the wrong order.

Reply:
as discussed, the status of a job will be finished updated by the orchestrator before handing it over to the dispatcher so no update to job status shall create a collision. The only scenario this might happen is when we try to cancel a job as this action will be trigger directly from the main thread while the job could be running in the subthread. But we have various conditions to check if we can cancel a job or not, if it's running then we can't, if it's not running, I think this process can still be triggered safely when we update the status of job to `canceled`